packages/utils/src/cache/cache-indexeddb.ts

```diff
@@ -0,0 +1,161 @@
+import { openDB } from 'idb';
+import type { DBSchema, IDBPDatabase } from 'idb';
+import type {
+  CacheStorage,
+  CacheItem,
+  CacheMap,
+  CacheIndexedDBStorageConfig
+} from './types';
+
+interface IndexedDBDatabase extends DBSchema {
+  [key: string]: {
+    key: string;
+    value: CacheItem;
+  }
+}
+
+const isFirefox = () => (/firefox/i).test(navigator.userAgent);
+
+/**
+ * Stores data in indexedDB for use across multiple sessions.
+ */
+export class CacheIndexedDBStorage implements CacheStorage {
+
+  /**
+   * A connection to a indexedDB database, use for open transaction or idb api
+   */
+  public db: IDBPDatabase<IndexedDBDatabase> | undefined;
+
+  /**
+   * Database name
+   */
+  public dbName: string;
+
+  /**
+   * Database Version.
+   */
+  public version: number;
+
+  /**
+   * Store name
+   */
+  public storeName: never;
+
+  /**
+   * Constructor
+   * @param info Storage config
+   */
+  constructor (info: CacheIndexedDBStorageConfig) {
+    const { dbName, version, storeName } = info;
+    this.dbName = dbName;
+    this.version = version;
+    this.storeName = storeName as never;
+  }
+
+  /**
+   * Set a item against a key to this storage
+   * @param key item key
+   * @param value item value
+   * @returns {void}
+   */
+  public async setItem (key: string, value: CacheItem): Promise<void> {
+    !this.db && await this.open();
+    const item = { ...value, key };
+    await this.db?.put(this.storeName, item, key);
+  }
+
+  /**
+   * Returns a item in this storage that matched by the key.
+   * @param key item key
+   * @returns cache item or null
+   */
+  public async getItem (key: string): Promise<CacheItem | null> {
+    !this.db && await this.open();
+    return await this.db?.get(this.storeName, key) || null;
+  }
+
+  /**
+   * Remove a item against a key to this storage
+   * @param key item key
+   * @returns {void}
+   */
+  public async removeItem (key: string): Promise<void> {
+    !this.db && await this.open();
+    await this.db?.delete(this.storeName, key);
```

Comment:
I'm not sure what will happen when call `removeItem` and key is not exist in store.
I think it will be better if `removeItem`  can return true or false, instead of void.
what do you think?